typescript/sdk/src/contracts/contracts.ts

```diff
@@ -231,6 +231,6 @@ export function appFromAddressesMapHelper<F extends HyperlaneFactories>(
 
   return {
     contractsMap: filteredContractsMap,
-    multiProvider: intersection.result,
+    multiProvider: multiProvider,
```

Comment:
Can you clarify why this was necessary? Why would having an MP with the same chains as the contracts map be an issue? There used to be places where the deployers/checkers/other infra stuff would loop based off the MP's chain list. Maybe that's no longer the case anywhere? If you're certain we only use the contracts/configs map for chains than its okay, just want to confirm. 

If you do want to keep it like this, no reason this can be just `multiProvider` instead of `mp: mp`

Reply:
yaa like eg. for routing ism deploy, {sepolia , arbsepolia, basesepolia} in config, but only {sepolia, arbsepolia} on contractMap so if we intersect MP and do .getDomainId for the basesepolia domain on sepolia, the deployer will be "no bueno"